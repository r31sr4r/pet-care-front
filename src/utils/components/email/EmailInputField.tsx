import { TextField } from "@mui/material";
import React, { useState } from 'react';

type Props = {
	emailValue: string;	
	handleEmailChange: (e: React.ChangeEvent<HTMLInputElement>) => void;	
};

export function EmailInputField({
	emailValue,	
	handleEmailChange,	
}: Props) {
	const [emailError, setEmailError] = useState(false);	

	const handleEmailValidation = () => {
		const emailRegex = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i;

		const emailValidation = emailRegex.test(emailValue);

		if (!emailValidation && emailValue) {
			setEmailError(true);
		} else {
			setEmailError(false);
		}
	};

	return (
		<TextField
			required
			fullWidth
			id="email"
			label="Email"
			name="email"
			autoComplete="email"
			value={emailValue}
			onChange={handleEmailChange}
			onBlur={handleEmailValidation}
			error={emailError}
			helperText={emailError ? 'Email invÃ¡lido.' : ''}
		/>
	);
}
