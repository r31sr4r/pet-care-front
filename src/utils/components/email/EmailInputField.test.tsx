import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { EmailInputField } from './EmailInputField';

const Props = {
    emailValue: '',
    handleEmailChange: () => {},
};

describe('EmailInputField', () => {
    it('should render successfully', () => {
        const { asFragment } = render(<EmailInputField {...Props} />, {
            wrapper: BrowserRouter,
        });
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render with a invalid email', () => {
        const { asFragment } = render(<EmailInputField {...Props} emailValue='mail@mail@.com'/>, {
            wrapper: BrowserRouter,
        });
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render with a valid email', () => {
        const { asFragment } = render(<EmailInputField {...Props} emailValue='mail@mail.com'/>, {
            wrapper: BrowserRouter,
        });
        expect(asFragment()).toMatchSnapshot();
    });
    
});