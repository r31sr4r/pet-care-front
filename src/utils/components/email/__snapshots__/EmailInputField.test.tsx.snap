// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInputField should render successfully 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for="email"
      id="email-label"
    >
      Email
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         *
      </span>
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        autocomplete="email"
        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id="email"
        name="email"
        required=""
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-yjsfm1"
        >
          <span>
            Email *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`EmailInputField should render with a invalid email 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="email"
      id="email-label"
    >
      Email
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         *
      </span>
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        autocomplete="email"
        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id="email"
        name="email"
        required=""
        type="text"
        value="mail@mail@.com"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            Email *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`EmailInputField should render with a valid email 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="email"
      id="email-label"
    >
      Email
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         *
      </span>
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        autocomplete="email"
        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id="email"
        name="email"
        required=""
        type="text"
        value="mail@mail.com"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            Email *
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
